snapcraft/extractors/appstream.py

```diff
@@ -194,6 +194,11 @@ def _get_icon_from_desktop_file(
         entry = DesktopEntry()
         entry.parse(os.path.join(workdir, path))
         icon = entry.getIcon()
+
+        if icon.startswith("${{SNAP}}/"):  # matches '${SNAP}'.
```

Comment:
I don't think this works, this is for formatting purposes.
If you want to be certain, then maybe do
```suggestion
        if icon.startswith(r"${SNAP}/"):
```